File: isprime.cpp
// isprime.cpp - 2016.08.02 - Atlee Brink

#include "isprime.hpp"

#include <iostream>

using namespace std;

using namespace nprime;

int main( int argc, char *argv[] )
{
  uint64_t num;

  if( argc != 2 || 1 != sscanf( argv[1], "%lu", &num ) ) {
    cout << "usage: " << argv[0] << " <uint64>" << endl;
    return 0;
  }

  cout << num << " is..." << flush;

  bool is = isprime( num );

  cout << "\r";

  cout << num << " is " << (isprime( num ) ? "" : "not ") << "prime" << endl;

  return 0;
}

File: javascript/isprime.js
// isprime.js - 2016.08.05 - Atlee Brink

function isPrime( num ) {

  if( num < 2 ) return false;
  if( num < 4 ) return true;
  if( num % 2 == 0 ) return false;

  const rt = Math.floor(Math.sqrt( num ));

  for( n = 3; n <= rt; n++ ) {
    if( num % n == 0 ) return false;
  }

  return true;
}
